src/main/java/seedu/address/model/residence/ResidenceName.java

```diff
@@ -3,22 +3,22 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import seedu.address.model.name.CommonName;
+
 /**
  * Represents a Residence's name in the ResidenceTracker.
  * Guarantees: immutable; is valid as declared in {@link #isValidResidenceName(String)}
  */
-public class ResidenceName {
+public class ResidenceName extends CommonName {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Residence Names should only contain alphanumeric characters spaces and '@', and it should not be blank";
```

Comment:
I think we can allow any special characters for ResidenceName because why restrict the user unnecessarily right? 

So a validity check could be to allow any non-empty strings and no "     "?